nomad/fsm_test.go

```diff
@@ -3305,6 +3305,36 @@ func TestFSM_DeleteNamespaces(t *testing.T) {
 	assert.Nil(out)
 }
 
+func TestFSM_DeleteNamespaces_ErrorSurfacing(t *testing.T) {
+	assert := assert.New(t)
+	ci.Parallel(t)
+	fsm := testFSM(t)
+
+	ns1 := mock.Namespace()
+	// force a failure by making this the default
+	ns1.Name = "default"
+	assert.Nil(fsm.State().UpsertNamespaces(1000, []*structs.Namespace{ns1}))
+
+	req := structs.NamespaceDeleteRequest{
+		Namespaces: []string{ns1.Name},
+	}
+
+	buf, err := structs.Encode(structs.NamespaceDeleteRequestType, req)
+	require.NoError(t, err)
+
+	resp := fsm.Apply(makeLog(buf))
+	if resp == nil {
+		t.Fatalf("no resp: %v", resp)
+	}
+	err, ok := resp.(error)
+	if !ok {
+		t.Fatalf("resp not of error type: %T %v", resp, resp)
+	}
+	if !strings.Contains(err.Error(), "default namespace can not be deleted") {
+		t.Fatalf("bad error: %v", err)
+	}
```

Comment:
We're idiomatically using `shoenig/test` unless there's a good reason not to. The assertions are typically shorter, and @shoenig has got a bunch of cool generics magic in there that makes the assertions more precise than those in testify. Not everywhere in the code base has been updated to that yet.

```suggestion
	must.NoError(t, err)

	resp := fsm.Apply(makeLog(buf))
	must.NotNil(t, resp)
	
	err, ok := resp.(error)
	must.True(t, ok, must.Sprintf("resp not of error type: %T %v", resp, resp)
	must.ErrorContains(t, err, "default namespace can not be deleted")
```

Reply:
Oooh that is way nicer